elastic/kibana
# [Discover][ES]QL] Ensure the same time range is being used for documents and histogram

## Summary

Resolves #201128

This PR fixes the case a relative time range is set (for example "Last 15 minutes") and the  time range goes out of sync for table and histogram requests on Discover in ES|QL mode. 

The main change is here:

```TypeScript
export interface DataMainMsg extends DataMsg {
   foundDocuments?: boolean;
   timeRange?: TimeRange;
   timeRangeRelative?: TimeRange;
 }
```

By extending the DataMainMsg with `timeRange`, `timeRangeRelative`, and consuming these values in subsequent code relevant for data fetching, we make sure, no relative timeRange is being converted with a delay causing a slightly different absolute timeRange. These values are set when the data fetching starts, and the way the code was rewritten also will lead to less redundant requests caused by the Histogram in Discover, but it doesn't fully resolve the issue #165192 (which I thought it does initially, built on the given approach to reduce the sources of truth for all relevant state for data fetching, I'm confident this can be resolved longer down the road)

**Testing**
1. Set a relative time range in Discover in ES|QL mode
2. Observe via Inspect that time range filter is the same for Table and Visualization calls.



### Checklist
